jme3-core/src/test/java/com/jme3/shader/GLSLPreprocessorTest.java

```diff
@@ -51,16 +51,16 @@
 public class GLSLPreprocessorTest {
 
     String readAllAsString(InputStream is) throws Exception{
-        String output = "";
+        StringBuilder output = new StringBuilder();
         BufferedReader reader = new BufferedReader(new InputStreamReader(is));
         while (true) {
             String l = reader.readLine();
             if (l == null) break;
-            if (output != "") output += "\n";
-            output += l;
+            if (output.toString() != "") output.append("\n");
```

Comment:
I think this code might be unreliable because it is using `!=` to compare string.
It was unreliable before this change, for the same reason, but obfuscating it in the name of performance just makes it worse.
Either keep it unchanged, or if you want to improve it, replace `output.toString() != ""` with `output.length() > 0`.

Reply:
Or if we are in the modern Java, then we can use isEmpty(). Can't remember when it came...

Reply:
As of Java 22, there's no `StringBuilder.isEmpty()`.
But `String.isEmpty()` dates back to Java 1.6, so that would work.
I just assume it would be more efficient to test the length property of the `StringBuilder` than to convert it to a `String` and then test *its* length property, regardless of whether you use `String.isEmpty()` or `String.length`.

Reply:
I think you are right about that.